pkgs/development/coq-modules/mathcomp/default.nix

```diff
@@ -4,24 +4,18 @@
 
 let params =
 
-  let param_1_7 = {
-      version = "1.7.0";
-      sha256 = "05zgyi4wmasi1rcyn5jq42w0bi9713q9m8dl1fdgl66nmacixh39";
-  }; in
-
+  if stdenv.lib.versionAtLeast coq.coq-version "8.6" then
   {
-    "8.5" =  {
-      version = "1.6.1";
-      sha256 = "1j9ylggjzrxz1i2hdl2yhsvmvy5z6l4rprwx7604401080p5sgjw";
-    };
-
-    "8.6" = param_1_7;
-    "8.7" = param_1_7;
-    "8.8" = param_1_7;
-    "8.9" = param_1_7;
+    version = "1.7.0";
+    sha256 = "05zgyi4wmasi1rcyn5jq42w0bi9713q9m8dl1fdgl66nmacixh39";
+  }
+  else if stdenv.lib.versionAtLeast coq.coq-version "8.5" then
+  {
+    version = "1.6.1";
+    sha256 = "1j9ylggjzrxz1i2hdl2yhsvmvy5z6l4rprwx7604401080p5sgjw";
+  }
+  else throw "No version of math-comp is available for Coq ${coq.coq-version}";
 
-  };
-  param = params."${coq.coq-version}";
```

Comment:
param is still beeing used in line 22.

Reply:
Although it should `fetchFromGitHub` instead of `fetchurl` since the archive's checksum can change if github decides to regenerate the archive.

Reply:
Thanks for noticing! Fixed (there is no `params` anymore). As for the `fetchurl` / `fetchFromGitHub` thing, I did not get the difference but in any case this is orthogonal to the current PR and can be changed independently.

Reply:
Added it because otherwise I would forget about that.